File: open_days_backend/src/main/java/com/sapientia/open/days/backend/ui/model/response/BaseErrorResponse.java
package com.sapientia.open.days.backend.ui.model.response;

/**
 * A basic model to return an error to the client. The BaseException uses this as return type.
 */
@SuppressWarnings("unused")
public class BaseErrorResponse {

	private int errorCode;
	private String errorMessage;

	public BaseErrorResponse() {
	}

	public BaseErrorResponse(int errorCode, String errorMessage) {
		this.errorCode = errorCode;
		this.errorMessage = errorMessage;
	}

	public int getErrorCode() {
		return errorCode;
	}

	public String getErrorMessage() {
		return errorMessage;
	}

	public void setErrorCode(int errorCode) {
		this.errorCode = errorCode;
	}

	public void setErrorMessage(String errorMessage) {
		this.errorMessage = errorMessage;
	}
}

File: open_days_frontend/lib/models/responses/base_error_response.dart
/// A basic model to represent an error from the server.
class BaseErrorResponse {
  int errorCode;
  String errorMessage;

  BaseErrorResponse({
    this.errorCode = -1,
    this.errorMessage = '',
  });

  factory BaseErrorResponse.fromJson(Map<String, dynamic> json) {
    return BaseErrorResponse(
      errorCode: json['errorCode'] ?? '',
      errorMessage: json['errorMessage'] ?? '',
    );
  }
}
